### ISSUE TITLE
Problem of network_loss

### ISSUE THREAD

**Issue by ponpon010548** [LTTM/IL-SemSegm#4]
Hi, I am trying to understand the function **network_loss** in _model_with_distillation.py_.
I'm confusing with the case **_applied_to == 'all',_** it seems to be the loss is applied to all classes, and the default _num_class  = 21_.

My question is why do we have to apply the loss to the class belonging to all of the unseen classes?
For example, we first train the model for 1_19, and we have default _num_class  = 21_. Thus, we are calculating the loss for 21 classes and we actually training on a model for 20 classes. 

In short, the point is whether this configuration is identical to the **equation (2)** in _Incremental Learning Techniques for Semantic Segmentation_.

Looking forward to your reply. Thanks!

**Comment by umbertomichieli** [comment]
Hi, you are correct. 
With 'all' it is applied to all the classes, however in the groundtruth there aren't future classes and so the results do not change masking out or not the following classes. While the second approach you are mentioning is more conceptualy correct, you can verify that the result is identical (I verified this time ago). You can easily verify this by applying a mask on the loss.
When I find the time I will commit also the other version.

So yes, you can think of that configuration as identical to equation (2).
Thank you.

**Comment by ponpon010548** [comment]
Thank you very much for your reply in detail! I will try it.
